resources/build.sh

```diff
@@ -296,27 +298,23 @@ echo ">> Configure boot FS"
 
 # download base firmware
 mkdir -p ${BOOTFS_PATH}
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/bootcode.bin
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup_cd.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup_db.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup_x.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup4.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup4cd.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup4db.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/fixup4x.dat
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start_cd.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start_db.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start_x.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start4.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start4cd.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start4db.elf
-wget -P ${BOOTFS_PATH} https://github.com/raspberrypi/firmware/raw/master/boot/start4x.elf
+git clone \
+    --bare --depth 1 \
+    --filter=blob:none --no-checkout \
+    https://github.com/raspberrypi/firmware ${BOOTFS_PATH}/.git
+
+(cd ${BOOTFS_PATH} \
+    && git fetch origin ${RPI_FIRMWARE_BRANCH} \
+    && git branch ${RPI_FIRMWARE_BRANCH} FETCH_HEAD \
+    && git --work-tree=${BOOTFS_PATH} checkout ${RPI_FIRMWARE_BRANCH} \
+    -- `git ls-tree refs/heads/${RPI_FIRMWARE_BRANCH} -r --name-only | grep -E "^boot\/([^\/]+)(\.dat|\.elf)$"`)
+
+mv ${BOOTFS_PATH}/boot/* ${BOOTFS_PATH}
+rm -rf ${BOOTFS_PATH}/boot ${BOOTFS_PATH}/.git
```

Comment:
Cool idea to git this way.

(I tried a complete shallow clone in the past but it took far to long)

Reply:
Totally agree! I think there's definitely a way to wrap this up much more slimmer, however I was surprised by the speed myself first.